shared-bindings/microcontroller/Pin.c

```diff
@@ -105,6 +105,42 @@ mcu_pin_obj_t *validate_obj_is_free_pin(mp_obj_t obj) {
     return pin;
 }
 
+// Validate every element in the list is a unique pin
+void validate_no_duplicate_pins(mp_obj_t seq, qstr arg_name) {
+    const size_t num_pins = (size_t)MP_OBJ_SMALL_INT_VALUE(mp_obj_len(seq));
+
+    for (size_t pin_cnt = 0; pin_cnt < num_pins; pin_cnt++) {
+        mp_obj_t pin1_obj = mp_obj_subscr(seq, MP_OBJ_NEW_SMALL_INT(pin_cnt), MP_OBJ_SENTINEL);
+        mcu_pin_obj_t *pin1 = validate_obj_is_pin(pin1_obj);
+
+        for (size_t pin_cnt_2 = pin_cnt + 1; pin_cnt_2 < num_pins; pin_cnt_2++) {
+            mp_obj_t pin2_obj = mp_obj_subscr(seq, MP_OBJ_NEW_SMALL_INT(pin_cnt_2), MP_OBJ_SENTINEL);
+            mcu_pin_obj_t *pin2 = validate_obj_is_pin(pin2_obj);
+            if (pin1 == pin2) {
+                mp_raise_TypeError_varg(translate("%q contains duplicate pins"), arg_name);
+            }
+        }
+    }
+}
+
+void validate_no_duplicate_pins_2(mp_obj_t seq1, mp_obj_t seq2, qstr arg_name1, qstr arg_name2) {
+    const size_t num_pins_1 = (size_t)MP_OBJ_SMALL_INT_VALUE(mp_obj_len(seq1));
+    const size_t num_pins_2 = (size_t)MP_OBJ_SMALL_INT_VALUE(mp_obj_len(seq2));
+
+    for (size_t pin_cnt_1 = 0; pin_cnt_1 < num_pins_1; pin_cnt_1++) {
+        mp_obj_t pin1_obj = mp_obj_subscr(seq1, MP_OBJ_NEW_SMALL_INT(pin_cnt_1), MP_OBJ_SENTINEL);
+        mcu_pin_obj_t *pin1 = validate_obj_is_pin(pin1_obj);
+
+        for (size_t pin_cnt_2 = 0; pin_cnt_2 < num_pins_2; pin_cnt_2++) {
+            mp_obj_t pin2_obj = mp_obj_subscr(seq2, MP_OBJ_NEW_SMALL_INT(pin_cnt_2), MP_OBJ_SENTINEL);
+            mcu_pin_obj_t *pin2 = validate_obj_is_pin(pin2_obj);
+            if (pin1 == pin2) {
+                mp_raise_TypeError_varg(translate("%q and %q contain duplicate objects"), arg_name1, arg_name2);
```

Comment:
Since this only checks pins, we can say "pins" instead of "objects"
```suggestion
                mp_raise_TypeError_varg(translate("%q and %q contain duplicate pins"), arg_name1, arg_name2);
```

Reply:
Good catch, missed it the second time.